tests/imputers/test_ArbitraryImputer.py

```diff
@@ -172,3 +175,20 @@ def test_super_columns_check_called(self, mocker):
         ):
 
             x.transform(df)
+
+    # Testing if the datatypes are preserved after imputation
+    def test_data_types_preserved(self):
```

Comment:
This test needs to to explicitly set the downcast dtypes.  Otherwise we need to fully test data_type_casting.py too. If that did not function as expected this test is compromised at present.